bin/all_sky_search/pycbc_fit_sngls_over_multiparam

```diff
@@ -245,13 +249,108 @@ assert len(args.log_param) == len(args.fit_param) == len(args.smoothing_width)
 
 init_logging(args.verbose)
 
-fits = h5py.File(args.template_fit_file, 'r')
+analysis_time = 0
+attr_dict = {}
+
+tid = numpy.array([], dtype=int)
+nabove = numpy.array([], dtype=int)
+ntotal = numpy.array([], dtype=int)
+alpha = numpy.array([], dtype=float)
+median_sigma = numpy.array([], dtype=float)
+
+logging.info("Loading input template fits")
+for filename in args.template_fit_file:
+    with h5py.File(filename, 'r') as fits:
+        # get attributes from the template-level fit
+        for k in fits.attrs.keys():
+            if k == 'analysis_time':
+                # For this, we want the mean, not check that it is identical
+                analysis_time += fits.attrs['analysis_time']
+                continue
+            if k not in attr_dict:
+                # This is the first time this attribute
+                # has been encountered
+                attr_dict[k] = fits.attrs[k]
+            elif not attr_dict[k] == fits.attrs[k]:
+                # Check that the files are consistent with one another
+                err_msg = f"Input files are not consistent, file {filename} "
+                err_msg += f"has attribute {k} of {fits.attrs[k]}, whereas "
+                err_msg += f"previous files have value {attr_dict[k]}"
+                raise RuntimeError(err_msg)
+
+        # get template id and template parameter values
+        tid = numpy.concatenate((tid, fits['template_id'][:]))
+        nabove = numpy.concatenate((nabove, fits['count_above_thresh'][:]))
+        ntotal = numpy.concatenate((ntotal, fits['count_in_template'][:]))
+        alpha = numpy.concatenate((alpha, fits['fit_coeff'][:]))
+        try:
+            median_sigma = numpy.concatenate((median_sigma, fits['median_sigma'][:]))
+        except KeyError:
+            logging.info('Median_sigma dataset not present in input file')
+            median_sigma = None
 
-# get the ifo from the template-level fit
-ifo = fits.attrs['ifo']
+# For an exponential fit 1/alpha is linear in the trigger statistic values
+# so calculating weighted sums or averages of 1/alpha is appropriate
+invalpha = 1. / alpha
+invalphan = invalpha * nabove
+
+# convert the sum above into a mean
+analysis_time /= len(args.template_fit_file)
+
+# If we are using multiple files, we should 'smooth' within each template
+# before smoothing over templates going forward. As the templates will
+# all have identical template_ids and have the same number of values,
```

Comment:
This explanation is not quite clear, which part of this calculation is 'taking the mean of means'?  The discussion also seems to mix up the implementation aspect (getting entries with identical tids) with the correctness of what is implemented. 

What I think this is saying is that out of the long array of n_templates * n_files values, averaging within each template involves getting the n_files occurrences which have the same tid, that explains the first part about implementation. 

Then you want to argue that the averaging here does not bias the final result. I am not sure if that _necessarily_ requires the same number of per-file entries for each template, but the linearity argument basically takes care of it